tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -1215,11 +1281,19 @@ class CudnnRNNForwardOp<GPUDevice, T> : public CudnnRNNKernelCommon {
           context, GetCachedRnnDescriptor<T>(context, model_shapes, input_mode,
                                              *output_algo_config,
                                              &rnn_state_cache_, &rnn_desc_ptr));
-      launch_status = DoForward<T>(
-          context, *rnn_desc_ptr, model_types(), model_shapes, input, input_h,
-          input_c, params, is_training_, output, output_h, output_c,
-          &reserve_space_allocator, &workspace_allocator,
-          /*output_profile_result=*/nullptr);
+      if (var_seq_lengths) {
+        launch_status = DoForward<T>(
+            context, *rnn_desc_ptr, model_types(), model_shapes, input, input_h,
+            input_c, params, is_training_, output, output_h, output_c,
+            &reserve_space_allocator, &workspace_allocator,
+            /*output_profile_result=*/nullptr, sequence_lengths);
```

Comment:
Move `sequence_lengths` before allocator params. Same for other places and backward.